File: chatbot.js
// JavaScript to handle chat interaction and communicate with Flask backend

function sendMessage() {
    const userInput = document.getElementById("user-input");
    const chatBox = document.getElementById("chat-box");
    const messageText = userInput.value.trim();
    
    if (messageText) {
        // Display user message in the chat box
        const userMessage = document.createElement("div");
        userMessage.className = "message user";
        userMessage.innerText = `You: ${messageText}`;
        chatBox.appendChild(userMessage);
        
        // Clear input field
        userInput.value = "";
        chatBox.scrollTop = chatBox.scrollHeight; // Keep scroll at the bottom
        
        // Send message to Flask backend and get bot response
        fetch('http://127.0.0.1:5000/chat', {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({ message: messageText })
        })
        
        .then(response => response.json())
        .then(data => {
            const botMessage = document.createElement("div");
            botMessage.className = "message bot";
            botMessage.innerText = `Bot: ${data.response}`;
            chatBox.appendChild(botMessage);
            
            // Scroll to show the latest bot response
            chatBox.scrollTop = chatBox.scrollHeight;
        })
        .catch(error => {
            console.error("Error:", error);
            const errorMessage = document.createElement("div");
            errorMessage.className = "message bot";
            errorMessage.innerText = "Bot: Sorry, I couldn't connect to the server.";
            chatBox.appendChild(errorMessage);
            chatBox.scrollTop = chatBox.scrollHeight;
        });
    }
}

// Optional: send message on Enter key press
document.getElementById("user-input").addEventListener("keypress", function(event) {
    if (event.key === "Enter") {
        sendMessage();
    }
});









// // JavaScript to handle chat interaction on the front end

// function sendMessage() {
//     const userInput = document.getElementById("user-input");
//     const chatBox = document.getElementById("chat-box");
//     const messageText = userInput.value.trim();

//     if (messageText) {
//         // Display user message
//         const userMessage = document.createElement("div");
//         userMessage.className = "message user";
//         userMessage.innerText = `You: ${messageText}`;
//         chatBox.appendChild(userMessage);

//         // Clear input field
//         userInput.value = "";

//         // Scroll to the bottom of the chat box to show the latest message
//         chatBox.scrollTop = chatBox.scrollHeight;

//         // Simulate a bot response after a short delay
//         setTimeout(() => {
//             const botMessage = document.createElement("div");
//             botMessage.className = "message bot";

//             // Sample bot responses based on user input (can be extended or replaced by backend responses)
//             if (messageText.toLowerCase().includes("hello") || messageText.toLowerCase().includes("hi")) {
//                 botMessage.innerText = "Bot: Hello! How can I assist you today?";
//             } else if (messageText.toLowerCase().includes("your name")) {
//                 botMessage.innerText = "Bot: I am PyBot, your virtual assistant!";
//             } else if (messageText.toLowerCase().includes("how are you")) {
//                 botMessage.innerText = "Bot: I'm just code, but I'm here to help you!";
//             } else if (messageText.toLowerCase().includes("bye")) {
//                 botMessage.innerText = "Bot: Goodbye! Have a great day!";
//             } else {
//                 botMessage.innerText = "Bot: I'm still learning. Could you please rephrase that?";
//             }

//             chatBox.appendChild(botMessage);
//             chatBox.scrollTop = chatBox.scrollHeight; // Keep scroll at the bottom
//         }, 500); // Simulate a short delay for bot response
//     }
// }

// // Optionally, add an event listener to send a message on Enter key press
// document.getElementById("user-input").addEventListener("keypress", function(event) {
//     if (event.key === "Enter") {
//         sendMessage();
//     }
// });

File: chatbot.py
from flask import Flask, request, jsonify
from flask_cors import CORS

app = Flask(__name__)
CORS(app)  # Enable CORS for your app

@app.route('/chat', methods=['POST'])
def chat():
    user_message = request.json.get("message")
    bot_response = ""
    
    if "bye" in user_message:
        bot_response = "Goodbye! Have a great day!"
    elif "hello" in user_message or "hi" in user_message or "hey" in user_message:
        bot_response = "Hello there! How can I help you today?"
    elif "your name" in user_message or "what is your name?" in user_message or "your name?" in user_message:
        bot_response = "I am a simple Python chatbot. You can call me PyBot!"
    elif "how are you" in user_message or "how are you?" in user_message:
        bot_response = "I'm just a bunch of code, but thank you for asking! How are you?"
    elif "time" in user_message:
        bot_response = "I don’t have a watch, but you can check it on your device!"
    else:
        bot_response = "I am still learning. Could you please rephrase that?"

    return jsonify({"response": bot_response})

# if __name__ == "__main__":
#     app.run(debug=True)


#_____________________________________________________________________________________________________________________________________________________

# # Simple Chatbot with continuous interaction using a while loop

# import flask
# from flask import Flask, request, jsonify
# from flask_cors import CORS
# # After initializing your app
# # CORS(chatbot)


# app = Flask(__name__)
# CORS(app)  # Enable CORS for your app

# @app.route('/chat', methods=['POST'])
# def chat():
#     user_message = request.json.get("message")

#     # Basic response logic - replace this with your chatbot's logic
#     # Checking if the user wants to end the conversation
#     if "bye" in user_input:
#         print("Chatbot: Goodbye! Have a great day!")
    
#     # Responses based on user input
#     elif "hello" in user_input or "hi" or "hey" in user_input:
#         print("Chatbot: Hello there! How can I help you today?")

#     elif "your name" or "what is your name?" or "your name?" in user_input:
#         print("Chatbot: I am a simple Python chatbot. You can call me PyBot!")

#     elif "how are you" or "how are you?" in user_input:
#         print("Chatbot: I'm just a bunch of code, but thank you for asking! How are you?")

#     elif "time" in user_input:
#         print("Chatbot: I don’t have a watch, but you can check it on your device!")

#     else:
#         print("Chatbot: I am still learning. Could you please rephrase that?")

#     return jsonify({"response": bot_response})

# if __name__ == "__main__":
#     app.run(debug=True)

